### ISSUE TITLE
[ObservableProperty] Mvvm SourceGenerator does not work for Generics

### ISSUE THREAD

**Issue by minesworld** [CommunityToolkit/dotnet#224]
### Describe the bug

```
public interface IX {
  public bool Value { get; }
}

public partial class X<P, M> : ObservableObject, IX {
  [ObservableProperty]
  private bool value;
}
```

will generate

```
public partial class X {
 [global::System.CodeDom.Compiler.GeneratedCode("Microsoft.Toolkit.Mvvm.SourceGenerators.ObservablePropertyGenerator", "7.1.0.0")]
        [global::System.Diagnostics.DebuggerNonUserCode]
        [global::System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage]
        public bool Value
        {
            get => value;
...
       }
}
```

and thus VisualStudio won't build the project as CS0535 error occurs:

`'X<P,M>' does not implement interface member IX.Value`

### Regression

7.1.2

### Reproducible in sample app?

- [ ] This bug can be reproduced in the sample app.

### Steps to reproduce

```text
- try the code with the Mvvm source generator
```

### Expected behavior

should generate the right code (whatever this is, I'm a C# newbie and not familar with partial classes)

### Windows Build Number

- [ ] Windows 10 1809 (Build 17763)
- [ ] Windows 10 1903 (Build 18362)
- [ ] Windows 10 1909 (Build 18363)
- [ ] Windows 10 2004 (Build 19041)
- [ ] Windows 10 20H2 (Build 19042)
- [ ] Windows 10 21H1 (Build 19043)
- [X] Windows 11 21H2 (Build 22000)
- [ ] Other (specify)

### App minimum and target SDK version

- [ ] Windows 10, version 1809 (Build 17763)
- [ ] Windows 10, version 1903 (Build 18362)
- [ ] Windows 10, version 1909 (Build 18363)
- [ ] Windows 10, version 2004 (Build 19041)
- [ ] Other (specify)

### Visual Studio Version

2022

### Visual Studio Build Number

17.1.4

### Device form factor

Desktop

### Nuget packages

Microsoft.Toolkit 7.1.2
Microsoft.Toolkit.Mvvm 7.1.2

### Help us help you

No.

**Comment by michael-hawker** [comment]
FYI @Sergio0694 (Also is there still some docs or something pointing to the main WCT repo, seeing a lot of issues still being opened up over there. Though I added a redirect entry in the new issue list recently too.)

**Comment by Sergio0694** [comment]
This is already fixed in main, and the fix is available in 8.0.0-Preview3.
